### PR TITLE
[Docs] Misc documentation fixes

### PR DESCRIPTION
**Pull request by foundrytom**

### CODE REVIEW

File: doc/src/Glossary.dox [hunk 1]
```diff
@@ -116,10 +116,10 @@
  *
  * The following are examples of perfectly valid entity references:
  *
- *  @li <tt>myasset://DVR/ep01/build/prop/table?v=6&t=model&f=alembic</tt>
- *  @li <tt>e1d2e41c-9594-11e2-9619-f23c91ae2917</tt>
- *  @li <tt>{'job':'DVR', 'path':['ep01','build','prop','table'],
- *  'version':'6', 'type':'model', 'format':'alembic'}</tt>
+ *  @li `myasset://DVR/ep01/build/prop/table?v=6&t=model&f=alembic`
```
Comment (feltech) [inline doc/src/Glossary.dox:119]: [fix] For some reason this file doesn't process the backticks

**Review by feltech — CHANGES_REQUESTED** [review]

**Comment by foundrytom** [comment]
@feltech Fixed a bunch of [invalid commands](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/94/commits/f76f8be9e843b58fd3d7ffc350d0c5df5c6efc33), that were breaking parsing, which seems to have fixed backticks in `Glossary`

File: doc/src/Glossary.dox (hunk 1 above)
Comment (foundrytom) [reply doc/src/Glossary.dox:119]: Fixed the file, but the last ref still didn't work with backticks so reverted [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/94/commits/01866e4f6b3b35fe28072ccc6386a01e4c323e68).

**Review by feltech — APPROVED** [review]

**Comment by feltech** [comment]
LGTM squash and merge when ready.

**Comment by feltech** [comment]
Frustratingly "feltech approved these changes" apparently before any commits were made. It looks like nothing has changed other than the `alt checkout` commit, which I assume is for testing CI. So I'll leave this PR assuming there's nothing to review for now.

**Comment by foundrytom** [comment]
> Frustratingly "feltech approved these changes" apparently before any commits were made. It looks like nothing has changed other than the `alt checkout` commit, which I assume is for testing CI. So I'll leave this PR assuming there's nothing to review for now.

Ahh sorry yes, this was a branch mixup 🙃, was leaving this PR open till #103 was merged!

**Comment by foundrytom** [comment]
> Ahh sorry yes, this was a branch mixup 🙃, was leaving this PR open till #103 was merged!

I've rebased this to a more modern base, and removed the extraneous commit.

**Pull request merged**